File: optimizer/src/optimizer/models/order.py
"""オーダーモデル — shared/types/order.ts に対応"""

from pydantic import BaseModel

from .common import DayOfWeek, ServiceType


class Order(BaseModel):
    id: str
    customer_id: str
    date: str  # "YYYY-MM-DD"
    day_of_week: DayOfWeek
    start_time: str  # "HH:MM"
    end_time: str  # "HH:MM"
    service_type: ServiceType
    staff_count: int = 1
    linked_order_id: str | None = None

File: shared/types/order.ts
import { Timestamp } from 'firebase-admin/firestore';
import { OrderStatus, ServiceType, TimeString } from './common';

/** オーダー（個別サービス依頼） */
export interface Order {
  id: string;
  customer_id: string;
  week_start_date: Timestamp;
  date: Timestamp;
  start_time: TimeString;
  end_time: TimeString;
  service_type: ServiceType;
  assigned_staff_ids: string[];
  staff_count?: number;
  status: OrderStatus;
  linked_order_id?: string;
  manually_edited: boolean;
  created_at: Timestamp;
  updated_at: Timestamp;
}
